File: src/main/java/com/example/domain/bookmark/entity/FestivalBookmark.java
package com.example.domain.bookmark.entity;

import com.example.domain.festival.entity.Festival;
import com.example.domain.member.entity.Member;
import com.example.global.entity.BaseCreatedEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_festival_bookmark_member_festival",
                        columnNames = {"member_id", "festival_id"}
                )
        }
)
public class FestivalBookmark extends BaseCreatedEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "festival_id", nullable = false)
    private Festival festival;

    //기본 생성자
    public FestivalBookmark() {
    }

    public Festival getFestival() {
        return festival;
    }

    public Member getMember() {
        return member;
    }

    //TODOS 생성자 구현
    public FestivalBookmark(Member member, Festival festival) {
        this.member = member;
        this.festival = festival;
    }
}

File: src/main/kotlin/com/example/domain/bookmark/entity/FestivalBookmark.kt
package com.example.domain.bookmark.entity

import com.example.domain.festival.entity.Festival
import com.example.domain.member.entity.Member
import com.example.domain.member.entity.QMember.member
import com.example.global.entity.BaseCreatedEntity
import jakarta.persistence.*

@Entity
@Table(
    uniqueConstraints = [UniqueConstraint(
        name = "uk_festival_bookmark_member_festival",
        columnNames = ["member_id", "festival_id"]
    )]
)
class FestivalBookmark(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    val member: Member,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "festival_id", nullable = false)
    val festival: Festival
) : BaseCreatedEntity() {}